spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -421,6 +421,78 @@ public void testSparkTableAddDropPartitions() throws Exception {
         "spark table partition should be empty", 0, sparkTable().partitioning().length);
   }
 
+  @Test
+  public void testUnboundPartitionSpecFormatVersion1() throws Exception {
+    sql(
```

Comment:
Just an idea: since both tests are identical except for the `format-version` property in the create statement, maybe we could combine them  and supply the format-version as a parameter in a loop? 
```
IntStream.rangeClosed(1, 2).forEach(version -> {
   ... // sql statements
});
```
This way it could be easily extended for future versions as well. WDYT?